internal/daemon/controller/handlers/hosts/host_service.go

```diff
@@ -773,11 +772,11 @@ func validateCreateRequest(req *pbs.CreateHostRequest) error {
 					len(attrs.GetAddress().GetValue()) > static.MaxHostAddressLength {
 					badFields[globals.AttributesAddressField] = fmt.Sprintf("Address length must be between %d and %d characters.", static.MinHostAddressLength, static.MaxHostAddressLength)
 				} else {
-					_, _, err := net.SplitHostPort(attrs.GetAddress().GetValue())
+					host, port, err := util.SplitHostPort(attrs.GetAddress().GetValue())
 					switch {
-					case err == nil:
+					case err == nil && port != "":
 						badFields[globals.AttributesAddressField] = "Address for static hosts does not support a port."
-					case strings.Contains(err.Error(), globals.MissingPortErrStr):
+					case (host != "" && port == "") || strings.Contains(err.Error(), globals.MissingPortErrStr):
```

Comment:
Should this not be
```suggestion
					case (host != "" && port == "") || strings.Contains(err.Error(), util.InvalidAddressContainsPort):
```
?
Also, since we own this function, can't we make it return a typed error instead of having to do this string matching? If it means changing how the function is used elsewhere too, I think that's worthwhile. We should use `errors.Is` where we can.

Reply:
> Should this not be ...

In this case, this switch case is not testing a negative case -- You can see it actually has no code that executes if it falls in this case. The reason this is here at all is due to the `default` case.

---

The logic around this is very convoluted and should be refactored. I have some ideas on improving all of this -- I'll be opening a PR and I'll tag you on it 😃.

Reply:
Sounds good, I'm happy to approve this pending that separate PR!